core/src/main/java/com/novoda/merlin/registerable/Registerer.java

```diff
@@ -5,79 +5,59 @@
-    private final MerlinConnector<Disconnectable> merlinDisconnector;
-    private final MerlinConnector<Bindable> merlinOnBinder;
+    private final Register<Connectable> connectables;
+    private final Register<Disconnectable> disconnectables;
+    private final Register<Bindable> bindables;
 
-    public Registerer(MerlinConnector<Connectable> merlinConnector, MerlinConnector<Disconnectable> merlinDisconnector, MerlinConnector<Bindable> merlinOnBinder) {
-        this.merlinConnector = merlinConnector;
-        this.merlinDisconnector = merlinDisconnector;
-        this.merlinOnBinder = merlinOnBinder;
+    public Registerer(Register<Connectable> connectables, Register<Disconnectable> disconnectables, Register<Bindable> bindables) {
+        this.connectables = connectables;
+        this.disconnectables = disconnectables;
+        this.bindables = bindables;
     }
 
     public void registerConnectable(Connectable connectable) {
-        getReconnector().register(connectable);
+        connectables().register(connectable);
     }
 
-    private MerlinConnector<Connectable> getReconnector() {
-        if (merlinConnector == null) {
+    private Register<Connectable> connectables() {
+        if (connectables == null) {
             throw new MerlinException(
                     "You must call " + Merlin.Builder.class.getSimpleName() + ".withConnectableCallbacks()" +
                             "before registering a " + Connectable.class.getSimpleName()
             );
         }
-        return merlinConnector;
+        return connectables;
     }
 
     public void registerDisconnectable(Disconnectable disconnectable) {
-        getDisconnector().register(disconnectable);
+        disconnectables().register(disconnectable);
     }
 
-    private MerlinConnector<Disconnectable> getDisconnector() {
-        if (merlinDisconnector == null) {
+    private Register<Disconnectable> disconnectables() {
+        if (disconnectables == null) {
             throw new MerlinException(
                     "You must call " + Merlin.Builder.class.getSimpleName() + ".withDisconnectableCallbacks()" +
                             "before registering a " + Disconnectable.class.getSimpleName()
             );
         }
-        return merlinDisconnector;
+        return disconnectables;
     }
 
     public void registerBindable(Bindable bindable) {
-        getOnBinder().register(bindable);
+        bindables().register(bindable);
     }
 
-    private MerlinConnector<Bindable> getOnBinder() {
-        if (merlinOnBinder == null) {
+    private Register<Bindable> bindables() {
+        if (bindables == null) {
             throw new MerlinException(
                     "You must call " + Merlin.Builder.class.getSimpleName() + ".withBindableCallbacks()" +
                             "before registering a " + Bindable.class.getSimpleName()
             );
         }
-        return merlinOnBinder;
-    }
-
-    public void register(Object object) {
```

Comment:
This was only used in tests. You could argue that this is exposed to the client but they have no way to add this to `Merlin` anyway so we can safely remove this.